util/train_test_functions.R

```diff
@@ -242,267 +242,142 @@ RestructureTrainedList <- function(train.list){
+    names(norm.list[[nrm.it]]) <- model.names
   }
   
-  train.kappa.mlt <- melt(train.kappa.list)
-  colnames(train.kappa.mlt) <- c("kappa", "perc.seq", "classifier", 
-                               "norm.method")
-  return(train.kappa.mlt)
-
-}
-
-PredictReconDataWrapper <- function(recon.list, train.list, sample.df,
-                                    return.kap = FALSE){
-  # This function is a wrapper for performing subtype prediction on 
-  # reconstructed test/hold-out data, using the models trained on training data
-  # in the supervised analysis (train.list from 2-train_test_brca_subtype.R).
-  # 
-  # Args:
-  #   recon.list: a list of reconstructed data
-  #   train.list: a list of predictive models (LASSO, linear SVM, random forest) 
-  #   sample.df: the data frame that maps sample name/header to subtype and
-  #              train/test set labels 
-  #              output of 0-expression_data_overlap_and_split.R
-  #   return.kap: logical; should the entire confusionMatrix (FALSE) or just
-  #               the Kappa statistic associated with the prediction?
-  # 
-  # Returns:
-  #   pred.list: a list of confusionMatrix objects (if return.kap = FALSE) 
-  #              or Kappa statistics (if return.kap = TRUE) from predictions on
-  #              the reconstructed data
-  #   
-  
-  pred.list <- list()  # initialize list for all predictions
-  for (mthd.iter in seq_along(train.list)) {  # for each normalization method
-    norm.list <- list()  # initialize list to hold all CM from all models
-    # and level of sequencing data
-    norm.mthd <- names(train.list)[mthd.iter] # name of normalization method
-    for (mdl.iter in 1:3) {  # for each model: glmnet, svm, rf
-      # excluding seeds element of list #4
-      mdl.list <- list()  #
-      mdl.name <- rownames(train.list[[norm.mthd]])[mdl.iter]  # name of model
-      for (seq.iter in 
-           seq_along(train.list[[mthd.iter]][[mdl.iter]])) {  # for each
-        # amount/level of sequencing data
-        seq.level <- names(train.list[[mthd.iter]][[mdl.iter]])[seq.iter]
-        # some methods (e.g., TDM) do not have data for each amount of
-        # sequencing, so check
-        if (!is.null(recon.list[[norm.mthd]][[seq.level]])) {
-          mdl <- train.list[[norm.mthd]][[mdl.iter]][[seq.level]]
-          recon.dt <- recon.list[[norm.mthd]][[seq.level]]
-          # get confusionMatrix
-          mdl.list[[seq.level]] <- PredictCM(model = mdl,
-                                             dt = recon.dt,
-                                             sample.df = sample.df,
-                                             model.type = mdl.name,
-                                             return.kappa = return.kap)
-        }
-      }
-      norm.list[[mdl.name]] <- mdl.list
-    }
-    pred.list[[norm.mthd]] <- norm.list
+  # if returning Kappa -- melt the list into a data.frame and return
+  # the data.frame
+  if (return.kap) {
```

Comment:
> when that's all that I would like returned in the majority of cases. 

gotchya, yeah, i see the argument for only returning kappa now. Perhaps then an a different solution in which you add a default argument in the wrapper function: `only.kappa = True`?

This way, you can set `only.kappa = False` and it returns both the confusionamatrix and kappa in a similar manner described above. Setting the default to true means that you won't have to update the function anywhere you use it only for kappa since it won't change that behavior